packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -304,8 +304,8 @@ function mountStatic(
 
     // slotAssignments can only apply to the top level element, never to a static part.
     patchSlotAssignment(null, vnode, renderer);
-    insertNode(elm, parent, anchor, renderer);
     mountStaticParts(elm, vnode, renderer);
+    insertNode(elm, parent, anchor, renderer);
```

Comment:
Why is it important to insert the node after `mountStaticParts`?

Reply:
In this PR it's not significant, however, when we're adding text nodes it may be more significant.

I planned to insert a space character in place of the dynamic text and I figured creating the element > mounting the parts > inserting the root node would prevent users from temporarily seeing the space character.

I'm open to setting it up either way though!

Reply:
For posterity, there was a significance in an edge case when mounting the parts before inserting into the dom mattered. 

See #4081 for details, it has been resolved in another PR.